addons/stock/models/stock_move.py

```diff
@@ -471,6 +482,32 @@ def action_show_details(self):
             ),
         }
 
+    def action_assign_serial_number(self):
+        """ Calls a wizard's method to assign SN's name on each move lines.
+        Returns an action to get back on the popup form view of detailed operations.
+        """
+        self.ensure_one()
+        if not self.next_serial_number_name:
+            raise UserError(_("You need to set a Serial Number before generating more."))
+        wiz = self.env['stock.assign.serial.numbers'].create({
+            'product_id': self.product_id.id,
+            'move_id': self.id,
+            'next_serial_number': self.next_serial_number_name,
+        })
+        wiz.generate_serial_numbers()
```

Comment:
why do you call it here?

Reply:
move _generate_serial_numbers to a private helper on stock move, use it here without the wizard stuff